server/app.js

```diff
@@ -1,17 +1,22 @@
 const express = require('express');
 const { getTopics } = require('./controllers/topics-controllers');
 const { getEndpoints } = require('./controllers/endpoint-controller');
+const { getArticleById } = require('./controllers/articles-controllers');
 const app = express();
 
 app.get('/api', getEndpoints);
 
 app.get('/api/topics', getTopics);
 
+app.get('/api/articles/:article_id', getArticleById);
+
 app.all('/*', (req, res, next)=> {
-    res.status(404).send({msg: 'Path not found'})
+    res.status(404).send({msg: 'Path not found'});
   });
 
 app.use((err, req, res, next) => {
-res.status(500).send({ msg: 'internal server error' })
+    if(err.status && err.msg) res.status(err.status).send({msg: err.msg});
+    if(err.code = 23502) res.status(400).send({msg:'Bad request'});
+    res.status(500).send({ msg: 'internal server error' });
 })
```

Comment:
These could be extracted out into an error handling controllers so you can have separate functions for `handleCustomErrors` and `handlePsqlErrors `